### PR TITLE
chore: improve python-progressbar maintenance path

### PR DESCRIPTION
**Pull request by lphuc2250gma**
Summary:
- Add narrow unit tests for edge cases in the utility helpers in progressbar/utils.py (e.g., len_color/no_color/AttributeBuffer behavior with empty strings, ANSI-only input, and mixed escape sequences) and tighten any missing type annotations on small helper functions. Purely additive tests plus minor type hint fixes — no behavior change.
- Keep the change narrow so it is straightforward to review.

Notes:
- I kept this scoped to the relevant implementation and tests.

### CODE REVIEW

File: progressbar/utils.py [hunk 1]
```diff
@@ -429,13 +429,13 @@ class AttributeDict(dict):
     AttributeError: No such attribute: spam
     """
 
-    def __getattr__(self, name: str) -> int:
+    def __getattr__(self, name: str) -> types.Any:
```
Comment (wolph) [inline progressbar/utils.py]: Why would we ever prefer `Any` over other types?

**Comment by wolph** [comment]
Thank you for this contribution, @lphuc2250gma! 🙏 The extra edge-case tests for `no_color`/`len_color`/`AttributeDict` are exactly the kind of concrete value assertions that keep utility helpers honest — several of them pin behaviors (bytes handling, ANSI-only input) that our coverage numbers alone never guaranteed. Your `AttributeDict` annotation fix had independently landed in our recent typing overhaul, so I merged develop into your branch and kept your tests (with two tiny lint tweaks). Much appreciated!
